plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1398,6 +1398,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
             transactionLogWriter.flush();
             writeCommitted = true;
             writeCheckpointIfNeeded(session, new SchemaTableName(handle.getSchemaName(), handle.getTableName()), checkpointInterval, commitVersion);
+
+            if (isCollectExtendedStatisticsColumnStatisticsOnWrite(session) && !computedStatistics.isEmpty() && !dataFileInfos.isEmpty()) {
+                // TODO: https://github.com/trinodb/trino/issues/16088 Add synchronization when version conflict for INSERT is resolved.
+                Optional<ExtendedStatistics> extendedStatistics = statisticsAccess.readExtendedStatistics(session, handle.getLocation());
+                Optional<Set<String>> analyzedColumns = extendedStatistics.flatMap(ExtendedStatistics::getAnalyzedColumns);
+                Optional<Instant> maxFileModificationTime = dataFileInfos.stream()
+                        .map(DataFileInfo::getCreationTime)
```

Comment:
I guess for `finishInsert` is not relevant, but eventually for `finishMerge` do take into account that you'll need to filter out only data files `dataFile.getDataFileType() == DATA`